cpp/ql/src/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -144,8 +144,23 @@ OutNode getAnOutNode(DataFlowCall call, ReturnKind kind) {
  */
 predicate jumpStep(Node n1, Node n2) { none() }
 
+/**
+ * Gets the field corresponding to the bit range `[startBit..endBit)` of class `c`.
```

Comment:
What does this do for the following cases:
```c++
struct SingleElement {
  int x;
};

struct Foo {
  int y;
  SingleElement z;
};

union U {
  int a;
  float b;
};

Foo s;
int w = s.z.x;  // startBit = 32 and endBit = 64, but does that mean "s.z.x" or "s.z"?
U u;
float f = u.b;  // startBit = 0 and endBit = 32, but does that mean "u.a" or "u.b"?
```

Reply:
In both cases it gives both results. Specifically, running
```codeql
from Class c, int start, int end, Field f
where f = getField(c, start, end)
select c, start, end, f
```
on your example gives:
```
| test.cpp:5:8:5:10 | Foo | 32 | 64 | test.cpp:2:7:2:7 | x |
| test.cpp:5:8:5:10 | Foo | 32 | 64 | test.cpp:7:17:7:17 | z |
| test.cpp:10:7:10:7 | U | 0 | 32 | test.cpp:11:7:11:7 | a |
| test.cpp:10:7:10:7 | U | 0 | 32 | test.cpp:12:9:12:9 | b |
```

Reply:
We don't want multiple results from `FieldContent.toString`. You could use `min` or `strictconcat` to collapse them down to one.

Reply:
Fixed in 01bf051.